cluster.go

```diff
@@ -954,14 +955,26 @@ func (c *Cluster) RecoverLocal(h *cid.Cid) (api.PinInfo, error) {
 // of the current global state. This is the source of truth as to which
 // pins are managed and their allocation, but does not indicate if
 // the item is successfully pinned. For that, use StatusAll().
-func (c *Cluster) Pins() []api.Pin {
+func (c *Cluster) Pins(quiet bool) []api.Pin {
 	cState, err := c.consensus.State()
 	if err != nil {
 		logger.Error(err)
 		return []api.Pin{}
 	}
+	pins := cState.List()
+	if !quiet {
+		return pins
+	}
+	// Filter out shard and clusterdag internal pins
+	quietPins := make([]api.Pin, 0)
+	for _, pin := range pins {
+		if pin.Type == api.CdagType || pin.Type == api.ShardType {
```

Comment:
As pointed above, let the filtering happen on the API only as it is purely a cosmetic feature for the moment.